src/solver/solvers.F90

```diff
@@ -1613,12 +1624,24 @@ subroutine convergenceInfo
                cgnsL2resK,    cgnsL2resOmega,   &
                cgnsL2resTau,  cgnsL2resEpsilon, &
                cgnsL2resV2,   cgnsL2resF        )
+#ifndef USE_COMPLEX
              monGlob(mm) = sqrt(monGlob(mm)/nCellGlobal(groundLevel))
+#else
+            ! take the square roots separately in complex mode
+            monGlob(mm) = cmplx(sqrt(real(monGlob(mm)/nCellGlobal(groundLevel))), &
+                               sqrt(aimag(monGlob(mm)/nCellGlobal(groundLevel))))
+#endif
+
              if (monNames(mm) == cgnsL2resRho) then
                 rhoRes = monGlob(mm)
              end if
           case ('totalR')
+#ifndef USE_COMPLEX
              monGlob(mm) = sqrt(monGlob(mm))
+#else
+             ! take the square roots separately in complex mode
+             monGlob(mm) = cmplx(sqrt(real(monGlob(mm))), sqrt(aimag(monGlob(mm))))
```

Comment:
We should consider adding a separate monitor for the norm of dR/dx.

As i mentioned on the issue you made for printing complex residuals. I don't think it is a good idea to compute h*|dR/dx|. It is best to first calculate the derivative and then take the norm of it.

Reply:
You just cannot square the complex residuals when they are that small, and need to scale them. You and I understand what derivative that ends up being, but for a general public, its just so much easier to have a simple L2 norm defined that they need to converge. For now, I am happy with what the code does and it works reliably for all of our current practices. I will leave that modification to you when you move the code to use 1e-200 by default.

Reply:
> You just cannot square the complex residuals when they are that small,

Yep I got that 

> For now, I am happy with what the code does and it works reliably for all of our current practices. I will leave that modification to you when you move the code to use 1e-200 by default.

sounds good